### DISCUSSION TITLE
DQN 코드 질문

### DISCUSSION THREAD

**Discussion by Lee-JaeWon** [reinforcement-learning-kr/2021_RLKR_Drone_Delivery_Challenge_with_Unity#9] [General]
제공된 DQN코드(`Agent_dqn.py`)에서 

```python
def convert_action(self, action):
        if action == 0:
            return [-1, -1, -1]
        elif action == 1:
            return [-1, -1,  0]
        elif action == 2:
            return [-1, -1,  1]
        elif action == 3:
            return [-1,  0, -1]
        elif action == 4:
            return [-1,  0,  0]
       # 이하 생략
```
가 무슨 역할을 하는건가요??

다른 간단한 예제에서 테스트해보려 코드를 고쳐보고 있는데
continuous한 action이 아니라 discrete action으로 다루려면 어떻게 해야하는지 모르겠습니다.ㅠㅠ

실행해보면
` the behavior needs a discrete input of dimension (1, 1) for (<number of agents>, <action size>) but received input of dimension (1, 3)` 과 같은 오류가 발생합니다.

```python
action_tuple = ActionTuple()
        action_tuple.add_discrete(np.array([action]))

        self.env.set_actions(self.behavior_name, action_tuple)
        self.env.step()
```
위 과정이 문제인듯합니다.

**Comment by LeejwUniverse** [comment]
안녕하세요. @Lee-JaeWon 님 대회 참여해주셔서 감사합니다.
기본적으로 Drone 환경의 경우 동시에 3가지의 continuous action을 step으로 받습니다.
[-1~1, -1~1, -1~1] 사이의 continuous action 값의 범위로 입력을 맞춰 주셔야 합니다.
만약 discrete action으로 접근하고 싶으시다면, convert_action 함수가 27개의 discrete action으로 임의로 선별한 함수 입니다.
baseline의 discrete action을 따를 필요도 없으며, 생각하시는 action을 추가하거나, action을 줄이셔도 됩니다.
